src/libraries/System.Runtime.InteropServices/gen/ComInterfaceGenerator/ComInterfaceAndMethodsContext.cs

```diff
@@ -0,0 +1,84 @@
+            /// </summary>
+            public IEnumerable<ComMethodContext> InheritedMethods => Methods.Where(m => m.DeclaringInterface != Interface);
+
+            public static IEnumerable<ComInterfaceAndMethodsContext> CalculateAllMethods(ValueEqualityImmutableDictionary<ComInterfaceContext, SequenceEqualImmutableArray<ComMethodInfo>> ifaceToDeclaredMethodsMap, StubEnvironment environment, CancellationToken ct)
+            {
+                Dictionary<ComInterfaceContext, (ImmutableArray<ComMethodContext> Methods, ImmutableArray<ComMethodContext> ShadowingMethods)> allMethodsCache = new();
+
+                foreach (var kvp in ifaceToDeclaredMethodsMap)
+                {
+                    AddMethods(kvp.Key, kvp.Value);
+                }
+
+                return allMethodsCache.Select(kvp => new ComInterfaceAndMethodsContext(kvp.Key, kvp.Value.Methods.ToSequenceEqual(), kvp.Value.ShadowingMethods.ToSequenceEqualImmutableArray()));
+
+                (ImmutableArray<ComMethodContext> Methods, ImmutableArray<ComMethodContext> ShadowingMethods) AddMethods(ComInterfaceContext iface, IEnumerable<ComMethodInfo> declaredMethods)
+                {
+                    if (allMethodsCache.TryGetValue(iface, out var cachedValue))
+                    {
+                        return cachedValue;
+                    }
+
+                    int startingIndex = 3;
+                    List<ComMethodContext> methods = new();
+                    // If we have a base interface, we should add the inherited methods to our list in vtable order
+                    if (iface.Base is not null)
+                    {
+                        var baseComIface = iface.Base;
+                        ImmutableArray<ComMethodContext> baseMethods;
+                        if (!allMethodsCache.TryGetValue(baseComIface, out var pair))
+                        {
+                            baseMethods = AddMethods(baseComIface, ifaceToDeclaredMethodsMap[baseComIface]).Methods;
+                        }
+                        else
+                        {
+                            baseMethods = pair.Methods;
+                        }
+                        methods.AddRange(baseMethods);
+                    }
+                    var shadowingMethods = methods.Select(method =>
+                    {
+                        var info = method.MethodInfo;
+                        var ctx = CalculateStubInformation(info.Syntax, info.Symbol, startingIndex, environment, iface.Info.Type, ct);
+                        return new ComMethodContext(iface, info, startingIndex++, ctx);
+                    }).ToImmutableArray();
+                    // Then we append the declared methods in vtable order
+                    foreach (var method in declaredMethods)
+                    {
+                        var ctx = CalculateStubInformation(method.Syntax, method.Symbol, startingIndex, environment, iface.Info.Type, ct);
```

Comment:
I'm not a fan of generating all of the methods at once here. I'd like to keep it such that changing the signature of one interface method doesn't require recalculation of the stub information for every single method on the interface and all derived interfaces.

Reply:
Changed this method to create a `ComMethodContext.Builder` that has everything except the value from `CalculateStubInformation`. Then another step maps over to call `CalculateStubInformation` and create `ComMethodContext`